islandora_pathauto.module

```diff
@@ -251,3 +253,87 @@ function islandora_pathauto_tokens($type, $tokens, array $data = array(), array
   }
   return array();
 }
+
+/*
+ * Implements hook_form_alter()
+ * 
+ * Adds a TextField to Islandora Properties so the User can Change
+ * the URL Alias.
+ */
+function islandora_pathauto_form_alter(&$form , &$form_state, $form_id){
+    if( $form_id == 'islandora_object_properties_form' ){
+        $generate_url = variable_get( 'set_generate', 0 );
+        
+        $form['check'] = array(
+            '#type'          => 'checkbox',
+            '#title'         => t('Generate automatic URL alias'),
+            '#default_value' => $generate_url,
+            '#description'   => t('Uncheck this to create a custom alias below.'),
+            '#weight'        => -1,
+        );
+        
+        $original_path = 'islandora/object/' . $form['pid']['#value'];
```

Comment:
Can current_path() be used here instead of hardcoding the string "islandora/object/" and the PID?